templates/pkg/resource/sdk.go.tpl

```diff
@@ -335,4 +335,7 @@ func (rm *resourceManager) handleImmutableFieldsChangedCondition(
 
 	return &resource{ko}
 }
-{{- end }}
\ No newline at end of file
+{{- end }}
+{{- if $hookCode := Hook .CRD "sdk_addon" }}
+{{ $hookCode }}
+{{- end }}
```

Comment:
Can you put this in a separate PR or commit ?

Reply:
agree with @A-Hilaly this should go in a separate PR (and I'm not sold on the idea of having a generic "sdk_addon" here. The hooks code was meant for code injection in specific places in the processing pipeline, not really as a way to add custom code outside of the processing pipeline...

Reply:
Created separate PR for it: https://github.com/aws-controllers-k8s/code-generator/pull/94